runner/interfaces.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2022 Cloudbase Solutions SRL
+//
+//    Licensed under the Apache License, Version 2.0 (the "License"); you may
+//    not use this file except in compliance with the License. You may obtain
+//    a copy of the License at
+//
+//         http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+//    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+//    License for the specific language governing permissions and limitations
+//    under the License.
+
+package runner
+
+import (
+	"context"
+	dbCommon "garm/database/common"
+	"garm/params"
+	"garm/runner/common"
+)
+
+type PoolManagerController interface {
+	CreateRepoPoolManager(ctx context.Context, repo params.Repository, providers map[string]common.Provider, store dbCommon.Store) (common.PoolManager, error)
+	GetRepoPoolManager(repo params.Repository) (common.PoolManager, error)
+	DeleteRepoPoolManager(repo params.Repository)
+	GetRepoPoolManagers() map[string]common.PoolManager
```

Comment:
Same as above.

Reply:
I don't believe this ever fail, because it only return the repo pool managers map (which is a field in the struct that implements the interface):
```
func (p *poolManagerCtrl) GetRepoPoolManagers() map[string]common.PoolManager {
	return p.repositories
}
```
This is only a `get` function for a private field.

Reply:
If this can have a different implementation at some point in the future, that needs to query the database of some external source, we'll have to change it then to add an error. It's ok to return ```nil``` as a value in the current implementation. This is an interface definition that can have multiple implementations. We need to be mindful of future needs so we don't add technical debt needlessly.